client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,131 @@
 import { emit } from 'api/common/socket';
-import { BoostConfigType } from 'types/boost';
+import {
+  FileConfigT,
+  ConfigBundleT,
+  ConfigObjRT,
+  ConfigT,
+  fileConfigTypeToRuntype,
+} from 'types/boost';
+import toast from 'react-hot-toast';
+import { Runtype, Static } from 'runtypes';
+
+type payloadActionT = 'upload' | 'delete';
+
+const configTypeToFileSuffix: { [K in ConfigT]: string } = {
+  rider: '_rider.json',
+  bike: '_bike.json',
+  track: '_track.json',
+  powerPlan: '_powerPlan.json',
+};
 
 /**
- * Send the content of the given configuration file on `boost/configs/action` over MQTT
+ * Alters the given string (by either adding a suffix or replacing it with '.json'). The suffix applied is determined by the config type provided.
  *
+ * @param name the string to be altered
+ * @param configType the type of config to determine the relevant suffix to use
+ * @param addSuffix true if the suffix should be added to the file name, false if it should be replaced with '.json' instead.
+ *
+ * @returns the given name altered
+ */
+export function alterFileSuffix(
+  name: string,
+  configType: ConfigT,
+  addSuffix: boolean,
+) {
+  if (addSuffix) {
+    return name.replace('.json', configTypeToFileSuffix[configType]);
+  }
+  return name.replace(configTypeToFileSuffix[configType], '.json');
+}
```

Comment:
Hate to be that guy, but this function seems to do two mostly unrelated things (indicated by having to pass in a boolean flag to decide which code to execute).

Having it as two functions without the `addSuffix` parameter would be nicer.